### DISCUSSION TITLE
How to avoid system boot from hanging when manually removing pools

### DISCUSSION THREAD

**Discussion by DotIN13** [openzfs/zfs#13611] [Q&A]
When I decided to take a pool out and put it elsewhere, the system would not boot unless I plug the pool back in. The boot messages hang at:

```
A start up job is running to import zfs pools from cache [1 min / no limit]
```

And as there is no limit, the system never boots. I am wondering if there is anything I could do to allow booting with missing cached pools.

I have tried appending `rootdelay` to grub but no avail. And I also realize that ZFS_INITRD_POST_MODPROBE_SLEEP='10' or ZFS_AUTOIMPORT_TIMEOUT='10' might help but it is also said that these were deprecated and removed so I have not tried those yet. Do we have any clue on whether they still work? Or any other way to achive that?

**Comment by rincebrain** [comment]
You've failed to mention any details which would allow us to look into what's going wrong, like your distro or version of ZFS.

I've not seen it hang indefinitely before - usually the import command will complain if something isn't there but was in the cachefile, but it shouldn't block boot, I believe...if it's doing that, then I suspect the import command is hanging on something, and the question becomes what. (The reason the question becomes that is, if it's blocking in the kernel, as I suspect it would be, then even if systemd wants to kill it it will fail until the command leaves the kernel.)

I wouldn't expect `rootdelay` or the like to help - those are for "the disks haven't appeared at the time the import was run, but will eventually", not "I have removed a pool and it will never come back".

**Reply by DotIN13** [reply]
I am running Manjaro `Linux Bumblebee 5.15.49-1-MANJARO #1 SMP PREEMPT Wed Jun 22 14:15:49 UTC 2022 x86_64 GNU/Linux` with the latest zfs `linux515-zfs 2.1.5-1.0`. But with previous versions I have also experienced the same issue so I assumed it was not version specific.

Regarding the information, is there any log device I am able to look at in order to understand what is blocking the boot?

**Reply by DotIN13** [reply]
I ran `journalctl -u zfs-import-cache` and got the following lines on boots that hang indefinitely.

```log
6月 30 11:42:05 Bumblebee systemd[1]: Starting Import ZFS pools by cache file...
6月 30 11:42:07 Bumblebee zpool[557]: cannot import 'bumblethings': no such pool or dataset
6月 30 11:42:07 Bumblebee zpool[557]: nvpair_value_nvlist(nvp, &rv) == 0 (0x16 == 0)
```
